### ISSUE TITLE
KeyError: 'PortfolioItem'

### ISSUE THREAD

**Issue by timrcase** [RallyTools/RallyRestToolkitForPython#105]
I'm getting a `KeyError: 'PortfolioItem'` when trying to work with User Stories. I've tried this:

```python
response = rally.get('UserStory', fetch=True)
for story in response:
    print(story.FormattedID)
    if story.PortfolioItem is not None:
        print(story.PortfolioItem._ref)
        print(story.PortfolioItem.FormattedID)
```

And I've also tried this:

```python
response = rally.get('UserStory', fetch=True, query='FormattedID = "US17058"')
story = response.next()
print story.details()
```

The `store.PortfolioItem._ref` line works but no matter what I do I cannot get to the FormattedID (or any other attribute of the Feature). Both of these return the following error.

```
Traceback (most recent call last):
  File "./rally-to-csv.py", line 19, in <module>
    print story.details()
  File "/usr/local/lib/python2.7/site-packages/pyral/entity.py", line 386, in details
    anv = "%s (OID  %s  FomattedID  %s)" % (anv, value.oid, value.FormattedID)
  File "/usr/local/lib/python2.7/site-packages/pyral/entity.py", line 150, in __getattr__
    item = response.content[rallyEntityTypeName]
KeyError: 'PortfolioItem'
```

**Comment by jfthuong** [comment]
I had a similar issue when trying to retrieve information about SagaFeature. I have found a way to patch it (I will enter a different issue on this topic).

In my case, the problem came from the computation of `rallyEntityTypeName` for a PortfolioItem (problem of casing): `entity_name, oid = self._ref.split(SLM_WS_VER)[-1].rsplit('/', 1)` in `entity.py`.

You can check the keys available in response.content by modifying `entity.py` by adding this line before the line `item = response.content[rallyEntityTypeName]`: 

`if 'Portfolio' in rallyEntityTypeName: print "\n\n".join(["%s => %s" % (k,v) for k,v in response.content.items()]) ##DEBUG`

And check what the key should be (instead of "PortfolioItem".

**Comment by klehman-rally** [comment]
TLDR;  specify your fetch fields as a list of Attributes, eg. fetch="Name,FormattedID,etc"

If you want to get at an entity's attribute sub-attribute (like the case here where you want to get the FormattedID of the Story's PortfolioItem attribute), you'll need to provide a fetch list (a string of comma separated Attribute ElementNames (ie., no spaces in the Attribute name and no space after each comma) including the Attribute ElementName for any Attribute sub-attribute you intend to reference.  In your example code, you have a fetch=True which will allow you to access each Attribute for the entity type you specify (ie., 'Story') but not for sub-objects (like PortfolioItem).  

There's indeed a problem with the details() method that will be fixed in v1.2.4. 

For requests that result in a large set of items, you are better off specifying all the Attributes you will be referencing as opposed to using fetch=True, especially if you end up using the details() method. 
The reason for that is when you request Story items with fetch=True and you use the details() method, it results in a separate request for the PortfolioItem (which does a fetch=True there also)  (let alone the cost of assembling _every_ field for every item on the server side as well as deconstructing it on the pyral side).  So instead of one or a few requests, you end up with essentially "number of qualifying items" requests which can get visibly tiresome looking at the staccato output on the console or waiting for the process to complete.
